src/topp/IDFilter.cpp

```diff
@@ -227,19 +229,16 @@ class TOPPIDFilter :
                                  "Similar to n_peptide_hits=1, but if there are ties between two or more highest scoring hits, none are kept.");
     registerStringOption_("best:n_to_m_peptide_hits", "[min]:[max]", ":", "Peptide hit rank range to extracts", false, true);
 
-    registerStringOption_("length", "[min]:[max]", ":", "Keep only peptide hits with a sequence length in this range.", false);
-
-    registerStringOption_("charge", "[min]:[max]", ":", "Keep only peptide hits with charge states in this range.", false);
 
     registerFlag_("var_mods", "Keep only peptide hits with variable modifications (as defined in the 'SearchParameters' section of the input file).", false);
 
-    registerFlag_("unique", "If a peptide hit occurs more than once per peptide ID, only one instance is kept.");
+    registerFlag_("unique_per_spectrum", "If a peptide-spectrum match occurs more than once for a spectrum, only one instance is kept.", true);
```

Comment:
Which one? First? Best by score?

Reply:
first occurrence (will add that see IDFilter.h comment).